packet.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"log"
+)
+
+const (
+	Data byte = iota
+	WindowResize
+)
+
+// arbitrary for now
+const PacketMaxSize int = 32
+
+type Packet struct {
+	Type byte
+	// number of bytes in the payload
+	Size    byte
+	Payload []byte
+}
+
+type PacketHandler func(ConnectionHandler, Packet)
+
+var HandlerTable = []PacketHandler{DataPacketHandler, WindowResizePacketHandler}
+
+func DataPacketHandler(handler ConnectionHandler, packet Packet) {
```

Comment:
Wouldn't it make more sense here to pass the connectionHandler pointer?

Reply:
In this case it's implicitly a `connectionHandler` pointer, if you trace back to `NewConnectionHandler` that returns `&h` (pointer to the `connectionHandler` struct it created), plus I guess since the interface methods are implemented as pointer-receiver methods on the struct then the compiler would complain otherwise? (an interesting read I found, tell me if I didn't understand it correctly though https://github.com/golang/go/wiki/MethodSets)
But yea agreed it's terribly confusing and maybe could be improved somehow